libsolidity/ast/Types.cpp

```diff
@@ -1569,12 +1569,15 @@ BoolResult ContractType::isImplicitlyConvertibleTo(Type const& _convertTo) const
 		return true;
 	if (_convertTo.category() == Category::Contract)
 	{
-		auto const& bases = contractDefinition().annotation().linearizedBaseContracts;
-		if (m_super && bases.size() <= 1)
+		auto const& targetContractType = dynamic_cast<ContractType const&>(_convertTo);
+		if (targetContractType.isSuper())
```

Comment:
I think we should also disallow conversions from super.

Reply:
It's already disallowed. The function has this at the beginning:
```c++
	if (isSuper())
		return nullptr;
```